odl/contrib/fom/supervised/supervised_fom.py

```diff
@@ -0,0 +1,570 @@
+        ``ref_data``. In normalized form the FOM takes values in [0, 1], with
+        higher correspondance at lower FOM value.
+    """
+
+    import odl
+
+    l2_normSquared = odl.solvers.L2NormSquared(test_data.space)
+
+    if mask is not None:
+        test_data = test_data * mask
+        ref_data = ref_data * mask
+
+    diff = test_data - ref_data
+    fom = l2_normSquared(diff)
+
+    if normalized:
+            fom /= (l2_normSquared(test_data) + l2_normSquared(ref_data))
+
+    return fom
+
+
+def mean_absolute_error(test_data, ref_data, mask=None, normalized=False):
+    """Mean absolute error-based FOM between ``test_data`` and ``ref_data``.
+
+    Evaluates mean absolute error between input (``test_data``) and
+    reference data (``ref_data``), by measuring consistency using the L1-norm.
+    Allows for normalization (``normalized``) and a masking of the two spaces
+    (``mask``).
+
+    Notes
+    ----------
+    The FOM evaluates
+
+    .. math::
+        \| f - g \|_1,
+
+    or, in normalized form
+
+    .. math::
+        \\frac{\| f - g \|_1}{\| f \|_1 + \| g \|_1}.
+
+    The normalized FOM takes its values in [0, 1], with higher correspondance
+    at lower FOM value.
+
+    Parameters
+    ----------
+    test_data : `LinearSpaceElement`
+        Input data or reconstruction.
+    ref_data : `LinearSpaceElement`
+        Reference data to compare ``test_data`` to.
+    mask : `LinearSpaceElement`
+        Binary mask to define ROI in which FOM evaluation is performed.
+    normalized  : `Boolean`
+        Boolean flag to switch between unormalized and normalized FOM.
+
+    Returns
+    -------
+    fom : `Scalar`
+        Scalar (float) indicating mean absolute error between ``test_data`` and
+        ``ref_data``. In normalized form the FOM takes values in [0, 1], with
+        higher correspondance at lower FOM value.
+    """
+
+    import odl
+
+    l1_norm = odl.solvers.L1Norm(test_data.space)
+    if mask:
+        test_data = test_data * mask
+        ref_data = ref_data * mask
+    diff = test_data - ref_data
+    fom = l1_norm(diff)
+
+    if normalized:
+        fom /= (l1_norm(test_data) + l1_norm(ref_data))
+
+    return fom
+
+
+def mean_value_difference(test_data, ref_data, mask=None, normalized=False):
+    """Mean value-based FOM between ``test_data`` and ``ref_data``.
+
+    Evaluates difference in mean value between input (``test_data``) and
+    reference data (``ref_data``). Allows for normalization (``normalized``)
+    and a masking of the two spaces (``mask``).
+
+    Notes
+    ----------
+    The FOM evaluates
+
+    .. math::
+         \\lvert  \\overline{f} - \\overline{g} \\rvert,
+
+    or, in normalized form
+
+    .. math::
+        \\bigg \\lvert \\frac{ \\overline{f} - \\overline{g} }
+                                 {\\overline{f} + \\overline{g}} \\bigg
+                                                                 \\rvert.
```

Comment:
This seems wrong. What if `g = -f`? Shouldn't it be absolute values of both `f` and `g`.

More generally, why would a normalized form make sense here?